tests/sqlparser_common.rs

```diff
@@ -8407,3 +8407,25 @@ fn test_buffer_reuse() {
     p.parse_statements().unwrap();
     let _ = p.into_tokens();
 }
+
+#[test]
+fn parse_json_table_function_err() {
+    let unsupported_dialects =
+        all_dialects_except(|d| d.is::<AnsiDialect>() || d.is::<MySqlDialect>());
+
+    // JSON_TABLE table function is not supported in the above dialects.
+    assert!(unsupported_dialects
+        .parse_sql_statements("SELECT * FROM JSON_TABLE('[[1, 2], [3, 4]]', '$[*]' COLUMNS(a INT PATH '$[0]', b INT PATH '$[1]')) AS t")
+        .is_err());
+}
+
+#[test]
+fn parse_json_table_as_identifier() {
+    let parsed = all_dialects().parse_sql_statements("SELECT * FROM json_table");
+    assert_eq!(
+        ParserError::ParserError(
+            "Cannot specify a keyword as identifier for table factor".to_string()
+        ),
+        parsed.unwrap_err()
+    );
```

Comment:
And, even for the supported dialects, `json_table` cannot be used as table identifier too. Previously it can be.